source/crud/crud.rst

```diff
@@ -1476,25 +1488,58 @@ Drivers SHOULD report errors however they report other server errors: by raising
 
 Drivers MUST construct a ``WriteConcernError`` from a server reply as follows:
 
-- set ``code`` to ``writeConcernError.code``.
-- set ``message`` to ``writeConcernError.errmsg`` if available.
-- set ``details`` to ``writeConcernError.errInfo`` if available. Drivers MUST NOT parse inside ``errInfo``.
+- Set ``code`` to ``writeConcernError.code``.
+- Set ``message`` to ``writeConcernError.errmsg``.
+- Set ``details`` to ``writeConcernError.errInfo`` if available. Drivers MUST NOT parse inside ``errInfo``.
+
+See the `Read/Write Concern specification </source/read-write-concern/read-write-concern.rst#writeconcernerror-examples>`_
+for examples of how a server represents write concern errors in replies.
+
+WriteError
+~~~~~~~~~~
+
+Write errors for ``insert``, ``update``, and ``delete`` commands are reported as
+objects within a ``writeErrors`` array field in the command response. Drivers
+MUST construct a ``WriteError`` from a server reply as follows (where
+``writeErrors[]`` refers to a particular element in the array):
+
+- Set ``code`` to ``writeErrors[].code``.
+- Set ``message`` to ``writeErrors[].errmsg`` if available.
+- Set ``details`` to ``writeErrors[].errInfo`` if available. Drivers MUST NOT parse inside ``errInfo``.
 
-See the `Read/Write Concern specification </source/read-write-concern/read-write-concern.rst#Errors>`_ for examples of how a server represents write concern errors in replies.
+For single-statement writes (i.e. ``insertOne``, ``updateOne``, ``updateMany``,
+``replaceOne``, ``deleteOne``, and ``deleteMany``), a single write error may be
+reported in the array and ``writeErrors[0].index`` will be zero.
+
+For multi-statement writes (i.e. ``insertMany`` and ``bulkWrite``), potentially
+many write errors may be reported in the array and the ``index`` property will
+be set accordingly. Since the reported ``index`` is specific to each command,
+drivers MUST adjust the index accordingly for ``BulkWriteError.index``.
```

Comment:
It seemed worth clarifying how we construct WriteError objects. Also, the spec didn't say much about how the `index` is calculated. This is another gap in the CRUD spec where the actual details are likely in the deprecated [Bulk API spec](https://github.com/mongodb/specifications/blob/master/source/driver-bulk-update.rst).